react/player/src/manager/managed-player.tsx

```diff
@@ -223,12 +235,24 @@ export const usePersistentStateMachine = (options: {
     _key: Symbol("managed-player"),
   });
 
-  const managedState =
-    managedPlayerStateMachines.get(keyRef.current) ??
-    new ManagedState({ middleware: options.middleware });
-  managedPlayerStateMachines.set(keyRef.current, managedState);
+  const initialManagedState = getOrCreateNewManagedState(
+    keyRef.current,
+    options.middleware,
+  );
+
+  const [managedState, setManagedState] = React.useState(initialManagedState);
```

Comment:
Do we need this kept in state, or could it be a `ref` instead?

Reply:
Since it should trigger a re-render when we create a new managed state I figured it should be kept in state. I guess technically because things will re-render anyways when the new flow comes in we don't need to do that twice. I can move it over to a ref.